tutorials/euclid_access/Euclid_ERO.md

```diff
@@ -0,0 +1,687 @@
+    phot2["gaia_{}".format(gaia_key)][sel_matched] = gaia_objects[gaia_key][idx[sel_matched]]
+```
+
+Once matched, we can now compare the Gaia and Euclid/NISP magnitudes of the stars.
+
+```{code-cell} ipython3
+# Data
+x = phot["gaia_phot_rp_mean_mag"]
+y = -2.5*np.log10(phot["flux_fit"]) + hdr["ZP_STACK"]
+
+# selection
+sel_good = np.where(phot["gaia_source_id"] > 0)[0]
+x = x[sel_good]
+y = y[sel_good]
+
+fig = plt.figure(figsize=(6,5))
+ax1 = fig.add_subplot(1,1,1)
+
+ax1.plot(x , y , "o", markersize=2)
+minlim = np.nanmin(np.concatenate((x,y)))
+maxlim = np.nanmax(np.concatenate((x,y)))
+
+ax1.fill_between(np.asarray([minlim,maxlim]),np.asarray([minlim,maxlim])/1.5,np.asarray([minlim,maxlim])*1.5, color="gray", alpha=0.2, linewidth=0)
+ax1.fill_between(np.asarray([minlim,maxlim]),np.asarray([minlim,maxlim])/1.2,np.asarray([minlim,maxlim])*1.2, color="gray", alpha=0.4, linewidth=0)
+ax1.plot(np.asarray([minlim,maxlim]),np.asarray([minlim,maxlim]), ":", color="gray")
+
+ax1.set_xlabel("Gaia Rp [mag]")
+ax1.set_ylabel("I$_E$ [mag]")
+plt.show()
+```
+
+## Visualization with Firefly
+
+At the end of this Notebook, we demonstrate how we can visualize the images and catalogs created above in `Firefly`.
+
+We start by initializing the Firefly client.
+The following line will open a new `Firefly` GUI in a separate tab **inside** the Jupyter Notebook environment. The user can drag the tab onto the currently open tab to create a "split tab". This the user to see the code and images side-by-side.
+
+```{code-cell} ipython3
+# Uncomment when using within Jupyter Lab with jupyter_firefly_extensions installed
+# fc = FireflyClient.make_lab_client()
+
+# Uncomment for contexts other than the above
+fc = FireflyClient.make_client(url="https://irsa.ipac.caltech.edu/irsaviewer")
+```
+
+In order to display in image or catalog in `Firefly`, it needs to be uploaded to the `Firefly` server. We do this here using the `upload_file()` function.
+We first upload the FITS image that we created above.
+
+```{code-cell} ipython3
+fval = fc.upload_file('./data/euclid_images_test.fits')
+```
+
+Once the image is uploaded we can use the `show_fits()` function to display it.
+Note that our FITS image has multiple extensions (VIS, and NISP bands). We can open them separately in new `Firefly` tabs by looping over the HDUs and specifying the plot ID by the extension's name.
+
+```{code-cell} ipython3
+for hh,hdu in enumerate(hdulcutout):
+    fc.show_fits(fval, MultiImageIdx=hh, plot_id=hdu.header["EXTNAME"] )
+```
+
+We can lock the WCS between the images (allowing the user to pan and zoom the images simultaneously) by running:
+
+```{code-cell} ipython3
+fc.align_images(lock_match=True)
+```
+
+In the same way, we can upload a table, in this case our Gaia table. We again use `upload_file()` but in this case we use `show_table()` to show it in `Firefly`.
+
+```{code-cell} ipython3
+tval = fc.upload_file('./data/gaiatable.csv')
+fc.show_table(tval, tbl_id = "gaiatable")
```

Comment:
Firefly supports out-of-the-box target based searches. I thought we have it documented but apparently not. Check `target_search_info` param in https://caltech-ipac.github.io/firefly_client/api/firefly_client.FireflyClient.html#firefly_client.FireflyClient.show_table

Since the Gaia table was already loaded, it's fine to save it and then load it to firefly.